py/client/pydeephaven/updateby.py

```diff
@@ -0,0 +1,239 @@
+            big_value_context (MathContext): the context to use when processing arbitrary precision numeric values,
+                default is DECIMAL128.
+        """
+        self.on_null = on_null
+        self.on_nan = on_nan
+        self.big_value_context = big_value_context
+
+    def make_grpc_request(self):
+        return GrpcUpdateByEmaOptions(on_null_value=self.on_null.value, on_nan_value=self.on_nan.value,
+                                      big_value_context=GrpcMathContext(precision=self.big_value_context.value[0],
+                                                                        rounding_mode=self.big_value_context.value[1]))
+
+
+class UpdateByOperation(UpdateByBase):
+    """A UpdateByOperation represents an operator for the Table update-by operation."""
+
+    def __init__(self, ub_column):
+        self.ub_column = ub_column
+
+    def make_grpc_request(self):
+        return GrpcUpdateByOperation(column=self.ub_column)
+
+
+def cum_sum(cols: List[str]) -> UpdateByOperation:
+    """ Create a cumulative sum update-by operation.
+
+    Args:
+        cols (List[str]): the columns to be operated on, can be a common name or an equal expression,
+                i.e. "col_a = col_b" for different column names; when empty, update_by performs the cumulative sum
+                operation on all the applicable columns.
+
+    Returns:
+        UpdateByOperation
+    """
+    ub_spec = GrpcUpdateBySpec(sum=GrpcUpdateBySpec.UpdateByCumulativeSum())
+    ub_column = GrpcUpdateByColumn(spec=ub_spec, match_pairs=cols)
+    return UpdateByOperation(ub_column=ub_column)
+
+
+def cum_prod(cols: List[str]) -> UpdateByOperation:
+    """ Create a cumulative product update-by operation.
+
+    Args:
+        cols (List[str]): the columns to be operated on, can be a common name or an equal expression,
+                i.e. "col_a = col_b" for different column names; when empty, update_by performs the cumulative product
+                operation on all the applicable columns.
+
+    Returns:
+        UpdateByOperation
+    """
+    ub_spec = GrpcUpdateBySpec(product=GrpcUpdateBySpec.UpdateByCumulativeProduct())
+    ub_column = GrpcUpdateByColumn(spec=ub_spec, match_pairs=cols)
+    return UpdateByOperation(ub_column=ub_column)
+
+
+def cum_min(cols: List[str]) -> UpdateByOperation:
+    """ Create a cumulative minimum update-by operation.
+
+    Args:
+        cols (List[str]): the columns to be operated on, can be a common name or an equal expression,
+                i.e. "col_a = col_b" for different column names; when empty, update_by performs the cumulative minimum
+                operation on all the applicable columns.
+
+    Returns:
+        UpdateByOperation
+    """
+    ub_spec = GrpcUpdateBySpec(min=GrpcUpdateBySpec.UpdateByCumulativeMin())
+    ub_column = GrpcUpdateByColumn(spec=ub_spec, match_pairs=cols)
+    return UpdateByOperation(ub_column=ub_column)
+
+
+def cum_max(cols: List[str]) -> UpdateByOperation:
```

Comment:
Server supports the union types for all of these: https://deephaven.io/core/pydoc/code/deephaven.updateby.html#deephaven.updateby.cum_sum